doc/api/vm.md

```diff
@@ -1573,6 +1573,77 @@ inside a `vm.Context`, functions passed to them will be added to global queues,
 which are shared by all contexts. Therefore, callbacks passed to those functions
 are not controllable through the timeout either.
 
+### Class: `NodeRealm`
+
+> Stability: 1 - Experimental
+
+<!-- YAML
+added: REPLACEME
+-->
+
+* Extends: {EventEmitter}
+
+A `NodeRealm` is effectively a Node.js environment that runs within the
+same thread.
+
+```mjs
+import { NodeRealm } from 'vm';
+import { fileURLToPath } from 'url';
+const w = new NodeRealm();
+const myAsyncFunction = w.createRequire(fileURLToPath(import.meta.url))('my-module');
+console.log(await myAsyncFunction());
+```
+
+#### `new NodeRealm()`
+
+<!-- YAML
+added: REPLACEME
+-->
+
+#### `noderealm.stop()`
+
+<!-- YAML
+added: REPLACEME
+-->
+
+This will render the inner Node.js instance unusable.
+and is generally comparable to running `process.exit()`.
+
+This method returns a promise that will be resolved when all resources
+associated with this Node.js instance are released. This promise resolves on
+the event loop of the _outer_ Node.js instance.
+
+#### `noderealm.createImport(filename)`
+
+<!-- YAML
+added: REPLACEME
+-->
+
+* `filename` {string}
+
+Create a dynamic `import()` function that can be used for loading EcmaScript
```

Comment:
`import()` is not a function, let’s remove that part as it’s confusing. IIUC it can be used to load ESM or CJS, so no need to specify ECMAScript either. 
```suggestion
Create a function that can be used for loading
```